shoppingcart/ui/main.go

```diff
@@ -0,0 +1,157 @@
+		"apple":      200,
+		"banana":     100,
+		"watermelon": 500,
+		"television": 100000,
+		"house":      100000000,
+		"car":        5000000,
+		"binder":     1000,
+	}
+	sessionId = newSession()
+)
+
+func main() {
+	var err error
+	workflowClient, err = client.Dial(client.Options{
+		HostPort: client.DefaultHostPort,
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	http.HandleFunc("/", listHandler)
+	http.HandleFunc("/action", actionHandler)
+
+	fmt.Println("Shopping Cart UI available at http://localhost:8080")
+	if err := http.ListenAndServe(":8080", nil); err != nil {
+		fmt.Println("Error starting server:", err)
+	}
+}
+
+func listHandler(w http.ResponseWriter, _ *http.Request) {
+	w.Header().Set("Content-Type", "text/html") // Set the content type to HTML
+	_, _ = fmt.Fprint(w, "<h1>SAMPLE SHOPPING WEBSITE</h1>"+
+		"<a href=\"/list\">HOME</a> <a href=\"/action?type=checkout\">Checkout</a>"+
+		"<h3>Available Items to Purchase</h3><table border=1><tr><th>Item</th><th>Cost</th><th>Action</th>")
+
+	keys := make([]string, 0)
+	for k := range itemCosts {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	for _, k := range keys {
+		actionButton := fmt.Sprintf("<a href=\"/action?type=add&id=%s\">"+
+			"<button style=\"background-color:#4CAF50;\">Add to Cart</button></a>", k)
+		dollars := float64(itemCosts[k]) / 100
+		_, _ = fmt.Fprintf(w, "<tr><td>%s</td><td>$%.2f</td><td>%s</td></tr>", k, dollars, actionButton)
+	}
+	_, _ = fmt.Fprint(w, "</table><h3>Current items in cart:</h3>"+
+		"<table border=1><tr><th>Item</th><th>Quantity</th><th>Action</th>")
+
+	cartState := updateWithStartCart("list", "")
+
+	// List current items in cart
+	keys = make([]string, 0)
+	for k := range cartState.Items {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	for _, k := range keys {
+		removeButton := fmt.Sprintf("<a href=\"/action?type=remove&id=%s\">"+
+			"<button style=\"background-color:#f44336;\">Remove Item</button></a>", k)
+		_, _ = fmt.Fprintf(w, "<tr><td>%s</td><td>%d</td><td>%s</td></tr>", k, cartState.Items[k], removeButton)
+	}
+	_, _ = fmt.Fprint(w, "</table>")
+}
+
+func actionHandler(w http.ResponseWriter, r *http.Request) {
+	actionType := r.URL.Query().Get("type")
+	switch actionType {
+	case "checkout":
+		err := workflowClient.SignalWorkflow(context.Background(), sessionId, "", "checkout", nil)
+		if err != nil {
+			log.Fatalln("Error signaling checkout:", err)
+		}
+		sessionId = newSession()
+		log.Println("Items checked out and workflow completed, starting new workflow")
+	case "add", "remove", "list":
+		id := r.URL.Query().Get("id")
+		updateWithStartCart(actionType, id)
+	default:
+		log.Fatalln("Invalid action type:", actionType)
+	}
+
+	// Generate the HTML after communicating with the Temporal workflow.
+	// "list" already generates HTML, so skip for that scenario
+	if actionType != "list" {
+		listHandler(w, r)
+	}
+}
+
+func updateWithStartCart(actionType string, id string) shoppingcart.CartState {
```

Comment:
WDYT about renaming this `id` variable to `itemID` everywhere in this file (incl. in the URL params)? That will help readers not get confused with the other IDs (workflow ID, and update ID; especially since workflow ID is simply called `ID` in the `StartWorkflowOptions` struct)